posix/stdlib.c

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/errno.h>
+#include <openenclave/corelibc/limits.h>
+#include <openenclave/corelibc/stdio.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/corelibc/string.h>
+#include <openenclave/corelibc/unistd.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/trace.h>
+
+char* oe_realpath(const char* path, char resolved_path[OE_PATH_MAX])
```

Comment:
>resolved_path[OE_PATH_MAX] [](start = 41, length = 26)

Replace function signatures using this construction with the standard `char *`; there was a discussion around this a while back and the consensus was that this provides no compile time checks and its irregular use was not sufficiently trusted as documentation (e.g. the `path` variable in this same signature is subject to the same constraints, but is not marked as such). In this specific case, you could pass in a buffer < OE_PATH_MAX and it would succeed assuming the resolved path was also less than that. 

On a separate note, we're using OE_PATH_MAX a lot internally in this set of changes, Windows still doesn't take relative paths > 260 and any extended paths taking the extended full path syntax from host can be ~32K (and only on NTFS) Have we accounted for that?

Reply:
I switched to using oe_path_t struct as the argument. I want to stick with the POSIX realpath() specification as much as possible, although I understand the need for compile time checking.

From the man page (for the corresponding **realpath** function):

```
       realpath() expands all symbolic links and resolves references  to  /./,
       /../  and  extra  '/' characters in the null-terminated string named by
       path to produce a canonicalized absolute pathname.  The resulting path‐
       name is stored as a null-terminated string, up to a maximum of PATH_MAX
       bytes, in the buffer pointed to by resolved_path.  The  resulting  path
       will have no symbolic link, /./ or /../ components.

       If  resolved_path  is specified as NULL, then realpath() uses malloc(3)
       to allocate a buffer of up to PATH_MAX bytes to hold the resolved path‐
       name,  and returns a pointer to this buffer.  The caller should deallo‐
       cate this buffer using free(3).
```

The PATH_MAX is a restriction that comes from the C library and from the definition of the function.